### DISCUSSION TITLE
Unable to bust cache of module federation remotes....

### DISCUSSION THREAD

**Discussion by shibisuriya** [webpack/webpack#18177] [Q&A]
## Feature request

**What is the expected behavior?**
I use dynamic import() to load the remoteEntry.js file (called moduleEntry.js in the screen shot)... Dynamic import is caching the remoteEntry.js file... I some how forced my app to load new instances of remoteEntry.js using cache busting... But I have no control over what the remoteEntry.js is further loading... The remoteEntry.js is further loading components code using dynamic import from the dev-server... But I can't bust this cache, so old components are getting loaded... 

![image](https://github.com/webpack/webpack/assets/69033713/a1189d6b-4fb2-49b2-9f3e-8c86e85344c1)

**What is motivation or use case for adding/changing the behavior?**
I am trying to build a dev tool that lets me develop components and embed inside a production webapp... So the remote will be running on my machine and the production app will be loaded from a cdn or something... These components that are loaded from the local, I need to support hot reloading inside the production app instead of refreshing the entire webapp... 

**How should this be implemented in your opinion?**
I don't have deep knowledge about webpack internals... A flag to cache bust not only the entry file but also the components that the entry file loads is good... 

**Are you willing to work on this yourself?**
yes

**Comment by alexander-akait** [comment]
Just use `[contenthash]` placeholder in a filename, so your filename will be unique

**Reply by shibisuriya** [reply]
@alexander-akait I have busted the cache by adding a query param in the url, for example, 127.0.0.1:8080/moduleEntry.js?&v=123,
I think modulEntry.js is not the problem... The installedChunk is the problem, the modules are cached there, how do I clear it? Thank you for helping.

![Uploading image.png…]()

**Reply by alexander-akait** [reply]
Please use `[contenthash]` to avoid caching in the `filename` option for the plugin, if you can provide your config and I will show
